[New] Add angular-mocks w/ npm auto-update

extend1994 opened the pull request:
Pull request for issue: #11160

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/angular/angular.js, https://github.com/angular/bower-angular-mocks
 * Official website (optional, not the repository): http://angularjs.org
 * NPM package url (optional): https://www.npmjs.com/package/angular-mocks
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 23
   - Count of stars: 92
   - Count of forks: 160
 * NPM download stats (optional):
   - Downloads in the last day: 21,143
   - Downloads in the last week: 110,808
   - Downloads in the last month: 508,656

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

Note:
I spared some spaces in .gitignore for avoiding any conflicts.

extend1994 commented:
@dakshshah96 @sufuf3 @hare1039 Please help me review this pull request, thank you.
Btw, I spared some spaces in .gitignore for avoiding any conflicts.

pvnr0082t commented on .gitignore:
```diff
@@ -205,6 +205,11 @@ node_modules/
+
+
+/ajax/libs/angular-mocks/*-build*/
+
```
Please remove unnecessary blank line. Thanks.

extend1994 replied:
@pvnr0082t Do you mean line 212?

sufuf3 commented on ajax/libs/angular-mocks/package.json:
```diff
@@ -0,0 +1,33 @@
+  ],
+  "author": "Angular Core Team <angular-core+npm@google.com>",
+  "license": "MIT",
+  "homepage": "http://angularjs.org",
```
Let's use `https`
https://angularjs.org/

sufuf3 commented:
I think `.gitignore` should use another commit. Because this will more clear that what commit do for what things.
What do you think?
ref: https://github.com/cdnjs/cdnjs/search?q=gitignore&type=Commits&utf8=%E2%9C%93

extend1994 commented:
@pvnr0082t @sufuf3 
I separated the previous commit into two commits, which are auto-update with the files and .gitconfig update respectively. And I modified the PR title which is better as Peter said. Last, I add https://github.com/angular/bower-angular-mocks in the comment but also reserved the origin one. Please help me review this pull request, thank you!

sufuf3 commented on .gitignore:
```diff
@@ -205,6 +205,9 @@ node_modules/
 
+
+/ajax/libs/angular-mocks/*-build*/
+
```
I think maybe we don't need to add the more blank line. L208 and L210

Just add `/ajax/libs/angular-mocks/*-build*/` between L206`/ajax/libs/europa/4.0.0-alpha*/` and L207 (blank line)

What do you think?

Thank you.

extend1994 replied:
@sufuf3 I add the blank lines for avoiding any conflicts because @PeterDaveHello told me that we can do that for conflicts. @PeterDaveHello Could you please help me confirm is these blank line alright? Thank you!

PeterDaveHello replied:
The empty lines shouldn't be added, and this may not help conflict parts at all.
The correct method to avoid the conflicts is to add the meaningful line randomly in the set, not just at the end of the set.

extend1994 commented:
@PeterDaveHello Sorry for misunderstanding. Now it should be fixed.
I randomly picked a line to insert `/ajax/libs/angular-mocks/*-build*/` which could help avoid conflicts.
@sufuf3 @pvnr0082t Please help me review this request, thank you.

PeterDaveHello commented:
Ping @maruilian11 @pvnr0082t

extend1994 commented:
ping @maruilian11 @pvnr0082t

extend1994 commented:
@maruilian11 @pvnr0082t Please help me review the pull request, thank you.

maruilian11 commented on ajax/libs/angular-mocks/package.json:
```diff
@@ -0,0 +1,33 @@
+  "description": "AngularJS mocks for testing",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/angular/angular.js.git"
```
Why don't we use `github.com/angular/bower-angular-mocks.git` as repo?

maruilian11 commented:
`*-build* versions of angular-mocks` in commit msg is not a sentence.

extend1994 commented:
@maruilian11 I updated the commit message. About the main reason why I didn't use github.com/angular/bower-angular-mocks.git as the repo is that I followed the upstream. The another reason is that I thought github.com/angular/bower-angular-mocks.git is the build version which is not exactly the same as the repo (but I am not sure about this. Please help me review this, thank you!

maruilian11 commented:
We can add versions before `1.2.18` from git repo. Using git auto-update will increase reviewer's burden.

extend1994 commented:
@maruilian11 I updated the repository info and changed the source of auto-update from GitHub to npm. Also, adding the files unavailable on npm from GitHub. Please help review this, thanks!
